File: scripts/pull-main.py
#!/usr/bin/env python3
import subprocess
import sys
import os

os.chdir('/vercel/share/v0-project')

try:
    print('[v0] Fetching latest changes from remote...')
    subprocess.run(['git', 'fetch', 'origin', 'main'], check=True)
    
    print('[v0] Current branch:')
    subprocess.run(['git', 'branch', '-v'], check=True)
    
    print('[v0] Pulling changes from main...')
    subprocess.run(['git', 'pull', 'origin', 'main'], check=True)
    
    print('[v0] Pull completed successfully')
    subprocess.run(['git', 'log', '--oneline', '-5'], check=True)
except subprocess.CalledProcessError as e:
    print(f'[v0] Error pulling changes: {e}', file=sys.stderr)
    sys.exit(1)

File: scripts/pull-main.mjs
import { execSync } from 'child_process';

console.log('[v0] Fetching latest changes from remote...');
try {
  execSync('git fetch origin main', { cwd: process.cwd(), stdio: 'inherit' });
  
  console.log('[v0] Current branch:');
  execSync('git branch -v', { cwd: process.cwd(), stdio: 'inherit' });
  
  console.log('[v0] Pulling changes from main...');
  execSync('git pull origin main', { cwd: process.cwd(), stdio: 'inherit' });
  
  console.log('[v0] Pull completed successfully');
  execSync('git log --oneline -5', { cwd: process.cwd(), stdio: 'inherit' });
} catch (error) {
  console.error('[v0] Error pulling changes:', error.message);
  process.exit(1);
}
